### PR TITLE
[release-v4.3] vendor: bump buildah to `v1.28.1`

### PR DESCRIPTION
**Pull request by flouthoc**
Contains backport of https://github.com/containers/buildah/pull/4411

[NO NEW TESTS NEEDED]
[NO TESTS NEEDED]

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by flouthoc** [comment]
@containers/podman-maintainers PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by flouthoc** [comment]
@giuseppe @vrothberg `/lgtm` was removed after my last push, I guess this needs `/lgtm` again.

**Review by vrothberg — COMMENTED** [review]
/lgtm

**Pull request merged**
